### PR TITLE
Bug/fix accounts credentials

### PR DESCRIPTION
**Pull request by einsteinx2**
**Is this pull request adding a feature or fixing a bug?**  
bug

**Does this pull request close an issue? If so, which one?**
#433

**What does this pull request do? What does it change?**
Fix moving old global credentials to new storage mechanism
Fix coinbase callback
Fix accounts not being unhidden when deleted
Fix crappy API code
Fix a small part of my sanity

### CODE REVIEW

File: Balance/Shared/Data Model/APIs/GDAX/GDAXAPIClient.swift [hunk 1]
```diff
@@ -244,71 +244,53 @@ extension GDAXAPIClient: ExchangeApi {
             } else if field.type == .passphrase {
                 passphrasField = field.value?.trimmingCharacters(in: .whitespacesAndNewlines)
             } else {
-                assert(false, "wrong fields are passed into the poloniex auth, we require secret and key fields and values")
+                assert(false, "wrong fields are passed into the gdax auth, we require secret and key fields and values")
             }
         }
         guard let secret = secretField, let key = keyField, let passphrase = passphrasField else {
-            assert(false, "wrong fields are passed into the poloniex auth, we require secret and key fields and values")
```
Comment (wolffan) [inline Balance/Shared/Data Model/APIs/GDAX/GDAXAPIClient.swift]: this was copy paste from polo? lol

File: Balance/Shared/Data Model/APIs/GDAX/GDAXAPIClient.swift [hunk 2]
```diff
@@ -320,7 +302,7 @@ extension GDAXAPIClient: ExchangeApi {
         catch {
             // TODO: show alert
             async {
-                closeBlock(false, error, nil)
+                closeBlock(false, error, institution)
```
Comment (wolffan) [inline Balance/Shared/Data Model/APIs/GDAX/GDAXAPIClient.swift]: wow 👍

Comment (wolffan) [reply Balance/Shared/Data Model/APIs/GDAX/GDAXAPIClient.swift]: btw this todo comment is a mistake?

File: Balance/Shared/Data Model/APIs/GDAX/GDAXCredentials.swift [hunk 3]
```diff
@@ -39,39 +40,37 @@ internal extension GDAXAPIClient
...
             }
             
             guard let unwrapedComponents = components else {
                 throw APICredentialsComponents.Error.dataNotFound(identifier: identifier)
             }
             
             try self.init(component: unwrapedComponents)
+            
+            // If the fetching of the old credentials succeeds, save the new ones and delete the old ones
+            if updatedCredentials {
```
Comment (wolffan) [inline Balance/Shared/Data Model/APIs/GDAX/GDAXCredentials.swift]: 👍

File: Balance/Shared/Data Model/APIs/GDAX/GDAXAPIClient.swift (hunk 2 above)
Comment (einsteinx2) [reply Balance/Shared/Data Model/APIs/GDAX/GDAXAPIClient.swift]: Probably, I wasn't 100% sure so I didn't remove it yet, but now that I think about it we do our own error UI using the error returned so it should be removed.

File: Balance/Shared/Data Model/APIs/GDAX/GDAXAPIClient.swift (hunk 1 above)
Comment (einsteinx2) [reply Balance/Shared/Data Model/APIs/GDAX/GDAXAPIClient.swift]: Haha yep

File: Balance/Shared/Data Model/APIs/GDAX/GDAXAPIClient.swift (hunk 2 above)
Comment (einsteinx2) [reply Balance/Shared/Data Model/APIs/GDAX/GDAXAPIClient.swift]: I'll remove it when I do the Kraken refactor

**Pull request merged**